compiler/rustc_span/src/lib.rs

```diff
@@ -1373,7 +1373,10 @@ impl Clone for SourceFile {
             },
             start_pos: self.start_pos,
             source_len: self.source_len,
-            lines: Lock::new(self.lines.borrow().clone()),
+            lines: {
+                let lock = self.lines.read();
+                FreezeLock::with(lock.clone(), self.lines.is_frozen())
```

Comment:
since there are two occurrences of this now, maybe add a non-self method to `FreezeLock` for cloning the lock?